### PR TITLE
Upgrade to scalatest 3.1.1

### PR DESCRIPTION
**Pull request by gheine**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -101,6 +107,7 @@ lazy val `scalatestplus-play` = project
       akkaHttpServer             % Test,
       "com.typesafe.play"        %% "play-test" % PlayVersion.current,
       "org.scalatest"            %% "scalatest" % ScalatestVersion,
+      "org.scalatestplus"        %% "selenium-2-45" % ScalatestSeleniumVersion,
```
Comment (raboof) [inline build.sbt:113]: The selenium version hinted at here (2.45) does not correspond with the selenium version we depend on (3.131.59). Is that a problem?

File: build.sbt [hunk 2]
```diff
@@ -117,7 +124,8 @@ lazy val docs = project
   .settings(
     commonSettings,
     libraryDependencies ++= Seq(
-      "org.mockito" % "mockito-core" % MockitoVersion % Test
+      "org.mockito"       % "mockito-core" % MockitoVersion          % Test,
+      "org.scalatestplus" %% "mockito-3-2" % ScalatestMockitoVersion % Test,
```
Comment (raboof) [inline build.sbt:131]: Here, too: we depend on 2.18.3, but I suspect this scalatestplus dependency pulls in 3.2 (though I haven't double-checked)

File: build.sbt (hunk 1 above)
Comment (gheine) [reply build.sbt:113]: 🤷 It doesn't look like it...

File: build.sbt (hunk 2 above)
Comment (gheine) [reply build.sbt:131]: Also doesn't appear to be a problem. Would suggest looking at mockito and selenium version alignment in a separate PR.

File: build.sbt (hunk 1 above)
Comment (gheine) [reply build.sbt:113]: Ideally this upgrade doesn't get blocked by https://github.com/scalatest/scalatestplus-selenium/pull/7

**Comment by gheine** [comment]
@raboof can this be released despite the version mismatch?

File: build.sbt (hunk 1 above)
Comment (raboof) [reply build.sbt:113]: The test failures in that PR suggest that upgrading selenium might not be without problems, even if we didn't see any in the test cases for this project. Before merging I'd like to be more sure that those are somehow not actually problems.

File: build.sbt (hunk 2 above)
Comment (raboof) [reply build.sbt:131]: that one seems safe: #253

File: build.sbt (hunk 1 above)
Comment (gheine) [reply build.sbt:113]: I don't think it's an actual problem. 

Scalatest 3.0.8, which scalatestplus-play 5.0.0 is using, was compiled against selenium 2.45.0 - see https://repo1.maven.org/maven2/org/scalatest/scalatest_2.13/3.0.8/scalatest_2.13-3.0.8.pom - while  scalatestplus-play 5.0.0 requires selenium 3.141.59 - see https://repo1.maven.org/maven2/org/scalatestplus/play/scalatestplus-play_2.13/5.0.0/scalatestplus-play_2.13-5.0.0.pom. 

The way I understand it is that from scalatest 3.0.8 to 3.1.1 the selenium specific code was just moved into a separate artifact (org.scalatestplus.selenium-2-45). There are no issues with the version mismatch or 5.0.0 has the same problems.

Comment (raboof) [reply build.sbt:113]: Ah, I didn't realize that - in that case there might still be a problem, but this PR doesn't make it worse, so :+1:

**Pull request merged**